dotnet/src/webdriver/BiDi/Modules/Network/UrlPattern.cs

```diff
@@ -22,27 +22,24 @@
 namespace OpenQA.Selenium.BiDi.Modules.Network;
 
 [JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
-[JsonDerivedType(typeof(Pattern), "pattern")]
-[JsonDerivedType(typeof(String), "string")]
+[JsonDerivedType(typeof(PatternUrlPattern), "pattern")]
+[JsonDerivedType(typeof(StringUrlPattern), "string")]
 public abstract record UrlPattern
 {
-    public static implicit operator UrlPattern(string value) => new String(value);
-
-    public record Pattern : UrlPattern
-    {
-        public string? Protocol { get; set; }
+    public static implicit operator UrlPattern(string value) => new StringUrlPattern(value);
```

Comment:
Passing null here can be dangerous

```suggestion
    [return: NotNullWhenNotNull(nameof(value))]
    public static implicit operator UrlPattern(string? value) => value is null ? null : new StringUrlPattern(value);
```

Reply:
We will review all implicit operators later, I am not even sure this approach is good.